veracode/submit_artifacts.py

```diff
@@ -386,51 +385,13 @@ def is_ready_for_new_build(*, upload_api: UploadAPI) -> bool:
             raise RuntimeError
 
         try:
-            # If a sandbox_id is already in progress, cancel it
-            sandbox_exists = response[0].get("sandbox_id")
-            if sandbox_exists is not None:
-                pass
-        except (KeyError, IndexError):
-            # No sandbox exists, no need to cancel existing builds
-            return True
-
-        # If the sandbox already exists OR this is a policy scan, see if a build is in-progress
-        endpoint = "getbuildinfo.do"
-        response = upload_api.http_get(endpoint=endpoint, params=params)
-
-        if element_contains_error(parsed_xml=response):
-            LOG.error("Veracode returned an error when attempting to call %s", endpoint)
-            raise RuntimeError
-
-        # Can only start a new build and perform file upload under the following two conditions
-        # https://help.veracode.com/reader/orRWez4I0tnZNaA_i0zn9g/KufwsiK88ub0VsFpNn8KcQ
-        try:
-            results_ready = response[0].get("results_ready")
-            if results_ready == "true":
-                return True
-        except (KeyError, IndexError):
-            LOG.error(
-                "Unable to extract the results_ready attribute from the Veracode response"
-            )
-            raise RuntimeError
+            # If we don't throw, at least one build already exists
```

Comment:
Are you expecting to throw an `IndexError`?  Because `KeyError` will never happen.
```python
>>> response = [{"not_build_id": "string"}]
>>> print(response[0].get("build_id"))
None
```

Reply:
Thanks, fixed.